### ISSUE TITLE
Speedlimit inside Geofence

### ISSUE THREAD

**Issue by Danteba** [traccar/traccar#3850]
Hello

I want to set a speed limit in a geofence and get notification if a vehicle exceeds the speed limit.

This can be useful because is not the same the speed limit in highways than in streets. I wish to make a geofence in the highway and put a higher speed limit in that zone.

Thank you 

Dante

**Comment by Danteba** [comment]
Another way to get similar results can be reached using the speedlimit information in openstreetmap.

**Comment by LucasBSC** [comment]
That is a nice feature. I am also interested on it. 
If anyone knows a way to get a similar result please tell me. 
Thank you,
Lucas

**Comment by Abyss777** [comment]
What would be suggestions about priority? Lets's say 
Device has speedLimit = 90 km/h and Geofence = 110 km/h ? 
What speed should generate event?

**Comment by LucasBSC** [comment]
I imagine that two separate different events should be generated.

**Comment by Danteba** [comment]
I think in different approachs:

a) The system launch two different kind of events:

1. Event launched when a device reached his own speedLimit
2. Event launched when a device reached a specific speedLimit inside a geofence

b) The geofence event takes priority over the device own speedLimit

**Comment by LucasBSC** [comment]
Agreed.

**Comment by Abyss777** [comment]
I would explain a bit, some time ago we are implemented tracing overspeeding as device state, to make it possible generate event with delay.

Tracing overspeeding states for every geofence and device itself is rather hard, and I do not think it should be implemented.

Second suggestion are closer to reality, but there is another problem, with delay now:

Let's say we have `event.overspeed.minimalDuration = 60` it is mean that device have to overspeed more than a minute to event be generated.

Geofence `City` has speedLimit 60 and geofence `Country` has speedLimit 90.

Imagine Driver near City border started overspeeding 60->95 about 15 seconds and when he/she get inside `Country` geofence he/she already overspeeding 95 and continue for 45 seconds. Should Event will be generated in this case?

In other words, changing speedLimit should reset device overspeeding state?

Another question what if device is inside two overlapped geofences in the same time which of two speedLimits should be used?

**Comment by LucasBSC** [comment]
Yes. I believe the event should be generated but that is why each event should be completely different from one another.

There would ideally be event for 'global' speedLimit and event for the speedLimit inside geofence separated from one another.

**Comment by Abyss777** [comment]
It would be ideal.
Frankly, I think it is not so valuable for now... If we could find some another feature to implement with "per geofence device states" we could combine development...

**Comment by LucasBSC** [comment]
Yes... I understand the effort to combine development.

I personally think it is a great to develop, I've seen some people looking for this feature here in my region since it is something that I've never seen implemented before but really plausible to have. It's up to you guys :D

**Comment by tananaev** [comment]
I just want to add that it's quite commonly requested feature. As for implementation, I think we should go with whatever is easier to implement. I don't think occasional "false" positive is a big problem for this feature.
